pennylane/operation.py

```diff
@@ -3034,6 +3035,57 @@ def convert_to_opmath(op):
     return op
 
 
+def convert_to_hamiltonian(op, wires_on_identity=True):
+    """
+    Converts arithmetic operators into :class:`~pennylane.Hamiltonian` instance.
+    Objects of any other type are returned directly.
+
+    Arithmetic operators include :class:`~pennylane.ops.op_math.Prod`,
+    :class:`~pennylane.ops.op_math.Sum` and :class:`~pennylane.ops.op_math.SProd`.
+
+    Args:
+        op (Operator): The operator instance to convert.
+        wires_on_identity (bool): If it is `True` preserves the wires associated with
+            the identity operator
+
+    Returns:
+        Operator: The operator as an :class:`~pennylane.Hamiltonian` instance
+    """
+    if isinstance(op, (qml.ops.op_math.Prod, qml.ops.op_math.SProd, qml.ops.op_math.Sum)):
+
+        op = qml.simplify(op)
```

Comment:
Since we are already doing this, we might as well just iterate through the operator and collect the items as they are, and simply transform Prods to Tensors (which is fair as I understand after simplification) - might save you the trouble of doing the Identity tracking.